### PR TITLE
Add User's and Developer's Guides to "docs"

### PR DESCRIPTION
**Pull request by kakirastern**
## Description

1. To add the User's Guide and Developer's guide to the `docs` subdirectory.
2. To add `sphinx-automodapi` to `glue-solar` in preparation of addition of `API` documentation.

### CODE REVIEW

File: docs/developers-guide.rst [hunk 1]
```diff
@@ -0,0 +1,209 @@
+.. _developers_guide:
+
+=================
+Developer's Guide
+=================
+
+How to Contribute to Glue-viz
+-----------------------------
+There are three ways by which anyone can contribute to the ``Glue-viz`` package and the ``Glue-solar``
```
Comment (nabobalis) [inline docs/developers-guide.rst]: I thought "Glue-viz" was the github org but the package was called "glue"?

File: docs/developers-guide.rst [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+plugin in particular is not only code (especially in Python), but also to report any issues you have
+encountered while using the software, helping to add or improve documentation, and to review a
+pull request (PR). All these will benefit other users of solar physics as they use ``Glue-viz`` for
+data visualization tasks, and help with the development of the community by offering them a valuable,
+powerful, and open-source tool to facilitate their research. Details about these routes of contributing
+are described below:
+
+Reporting Issues
+^^^^^^^^^^^^^^^^
+If you use ``Glue`` with the ``Glue-viz`` plugin and encounter any problems, the best way to report it is
```
Comment (nabobalis) [inline docs/developers-guide.rst]: "Glue-viz" is a plugin?

File: docs/developers-guide.rst [hunk 3]
```diff
@@ -0,0 +1,209 @@
+.. _developers_guide:
+
+=================
+Developer's Guide
+=================
+
+How to Contribute to Glue-viz
+-----------------------------
+There are three ways by which anyone can contribute to the ``Glue-viz`` package and the ``Glue-solar``
+plugin in particular is not only code (especially in Python), but also to report any issues you have
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
plugin in particular is not only code, but also to report any issues you have
```

File: docs/developers-guide.rst [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+encountered while using the software, helping to add or improve documentation, and to review a
+pull request (PR). All these will benefit other users of solar physics as they use ``Glue-viz`` for
```
Comment (nabobalis) [inline docs/developers-guide.rst]: Should "Glue-viz" here be "glue-solar"?

File: docs/developers-guide.rst [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+data visualization tasks, and help with the development of the community by offering them a valuable,
+powerful, and open-source tool to facilitate their research. Details about these routes of contributing
+are described below:
+
+Reporting Issues
+^^^^^^^^^^^^^^^^
+If you use ``Glue`` with the ``Glue-viz`` plugin and encounter any problems, the best way to report it is
+by opening an issue on our GitHub issue tracker. That way we can help you work around the problem
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
by opening an issue on our GitHub issue tracker. This way, we can help you work around the problem
```

File: docs/developers-guide.rst [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+and hopefully resolve it as soon as it arises. To use GitHub, you will better to have opened a GitHub account
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
and hopefully resolve it as soon as it arises. To use GitHub, you will will need to have a GitHub account
```

File: docs/developers-guide.rst [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+and signed into it to report an issue. Having joined GitHub will make it easier to report and track
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
and be signed into it to report an issue. Having joined GitHub will make it easier to report and track
```

File: docs/developers-guide.rst [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+provide a small code sample for us to attempt to reproduce the error. Also, record any error output generated
+when you stumbled upon the issue, so we can use this information to patch the codebase.
+
+If there is functionality that is not currently available in ``Glue-viz`` or ``Glue-solar`` you can make
+a feature request. Please provide as much information as possible regarding the feature you would like to see in
+``Glue-solar``.
+
+Documentation
+^^^^^^^^^^^^^
+``Glue-solar`` has some `online documentation <http://glue-solar.rtfd.io/>`_ and we strive to make it as comprehensive
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
``Glue-solar`` has some `online documentation <https://glue-solar.readthedocs.io/en/latest/>`__ and we strive to make it as comprehensive
```
Any place we use the old URL should be changed.

File: docs/developers-guide.rst [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+as possible. This documentation contains the API of ``Glue-solar`` but also a user's guide
+and developer's documentation.
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
and developer documentation.
```

File: docs/developers-guide.rst [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+Either the ``Glue-viz`` or ``Glue`` repositories are good staring points to investigate. The primary one is the
+``Glue-viz`` repository with where all the known issues for the core package are detailed. Each issue
+should have a series of labels that provide some information regarding its nature. If you find an issue
+that you would like to work on, please make sure to add a comment to let people know that you are working on it.
+This will ensure that it is less likely similar effort is duplicated by multiple parties.
+
+In addition, you can also explore the repository of the ``Glue-solar`` plugin where you are encouraged to find
+open issues there, and you might find these more interesting than those in the main ``Glue-viz`` repository.
+
+
```
Comment (nabobalis) [inline docs/developers-guide.rst:62]:
```suggestion
```

File: docs/developers-guide.rst [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+----------------
+The purpose of this statement is to set forth the standards that are expected of all code contributions to
+``Glue-solar``. All potential developers should read and abide by the following standards as much as possible.
+Code which does not follow these standards closely will not be accepted.
+
+Language Standard
+^^^^^^^^^^^^^^^^^
+1. All code must be compatible with Python 3.7 and later. Usage of ``six``, ``__future__``, and ``2to3``
+is not acceptable.
+2. The new Python 3 formatting style should be used throughout; i.e., use ``"{0:s}".format("spam")``
```
Comment (nabobalis) [inline docs/developers-guide.rst]: Should probably be updated to use f-strings instead of format.

File: docs/developers-guide.rst [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+These examples should be short and simple to reproduce, so that users would be able to copy them verbatim
+and run them. These examples should also, whenever possible, be in the
+`doctest <https://docs.astropy.org/en/stable/development/testguide.html#doctests>`_
+format and will be executed as part of the test suite.
+4. Unit tests should be provided for as many public methods and functions as possible.
+
+Glue-solar Documentation Rules
+------------------------------
+We recommend following for example the
+`SunPy Documentation Rules <https://docs.sunpy.org/en/latest/dev_guide/documentation.html>`_, though it is not
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
`SunPy Documentation Rules <https://docs.sunpy.org/en/latest/dev_guide/documentation.html>`__, though it is not
```

File: docs/developers-guide.rst [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+Documentation and Testing
+^^^^^^^^^^^^^^^^^^^^^^^^^
+1. American English is the default language for all documentation strings and inline commands.
+Variables names should also be based on English words as far as possible.
+2. Documentation strings must be present for all public classes/methods/functions. Furthermore, inclusion of
+examples or tutorials in the package documentation is strongly recommended and encouraged.
+3. Write usage examples in the docstrings of all classes and functions whenever possible.
+These examples should be short and simple to reproduce, so that users would be able to copy them verbatim
+and run them. These examples should also, whenever possible, be in the
+`doctest <https://docs.astropy.org/en/stable/development/testguide.html#doctests>`_
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
`doctest <https://docs.astropy.org/en/stable/development/testguide.html#doctests>`__
```

File: docs/developers-guide.rst [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+3. Following PEP8’s recommendation, absolute imports are to be used in general. Relative imports within a module are
+allowed to avoid circular import chains.
+4. The ``import numpy as np``, ``import matplotlib as mpl``, and ``import matplotlib.pyplot as plt`` naming conventions
+for import statements should be used wherever and whenever relevant. However, ``from packagename import *`` should
+never be used, expect in ``__init__.py``.
+5. Classes should either use direct variable access, or Python’s property mechanism for setting
+object instance variables.
+6. Classes should use the builtin `super() <https://docs.python.org/3/library/functions.html#super>`_ function
+when making calls to methods in their super-class(es), unless there are specific reasons not to. Also,
+`super() <https://docs.python.org/3/library/functions.html#super>`_ should be used consistently in all subclasses,
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
`super() <https://docs.python.org/3/library/functions.html#super>`__ should be used consistently in all subclasses,
```

File: docs/developers-guide.rst [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+2. **Follow the existing coding style** within a file and avoid making changes that are purely stylistic.
+Please try to maintain the style when adding or modifying code.
+3. Following PEP8’s recommendation, absolute imports are to be used in general. Relative imports within a module are
+allowed to avoid circular import chains.
+4. The ``import numpy as np``, ``import matplotlib as mpl``, and ``import matplotlib.pyplot as plt`` naming conventions
+for import statements should be used wherever and whenever relevant. However, ``from packagename import *`` should
+never be used, expect in ``__init__.py``.
+5. Classes should either use direct variable access, or Python’s property mechanism for setting
+object instance variables.
+6. Classes should use the builtin `super() <https://docs.python.org/3/library/functions.html#super>`_ function
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
6. Classes should use the builtin `super() <https://docs.python.org/3/library/functions.html#super>`__ function
```

File: docs/developers-guide.rst [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+
+Language Standard
+^^^^^^^^^^^^^^^^^
+1. All code must be compatible with Python 3.7 and later. Usage of ``six``, ``__future__``, and ``2to3``
+is not acceptable.
+2. The new Python 3 formatting style should be used throughout; i.e., use ``"{0:s}".format("spam")``
+instead of ``"%s" % "spam"``.
+3. The ``Glue`` package and the ``Glue-solar`` plugin should be importable with no dependencies other than those
+already in ``Glue`` and ``Glue-solar``, the
+`Python Standard Library <https://docs.python.org/3/library/index.html>`_, and packages that are already
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
`Python Standard Library <https://docs.python.org/3/library/index.html>`__, and packages that are already
```

File: docs/developers-guide.rst [hunk 17]
```diff
@@ -0,0 +1,209 @@
...
+required to follow the rules strictly as we prefer to use these as guidelines rather than rules.
+
+Overview
+^^^^^^^^
+For the RST files, we recommend a one-sentence-per-line rule and ignore the line length.
+
+Sphinx
+^^^^^^
+All of the ``Glue-solar`` documentation (like this page) is built with
+`Sphinx <https://www.sphinx-doc.org/en/stable/>`_, which is a tool very suitable for documenting Python projects.
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
`Sphinx <https://www.sphinx-doc.org/en/stable/>`__, which is a tool very suitable for documenting Python projects.
```

File: docs/developers-guide.rst [hunk 18]
```diff
@@ -0,0 +1,209 @@
...
+Sphinx works by parsing files written in a
+`Media-wiki-like syntax <http://docutils.sourceforge.net/docs/user/rst/quickstart.html>`_ called
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
`Media-wiki-like syntax <http://docutils.sourceforge.net/docs/user/rst/quickstart.html>`__ called
```

File: docs/developers-guide.rst [hunk 19]
```diff
@@ -0,0 +1,209 @@
...
+`reStructuredText <http://docutils.sourceforge.net/rst.html>`_. In addition to parsing static files
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
`reStructuredText <http://docutils.sourceforge.net/rst.html>`__. In addition to parsing static files
```

File: docs/developers-guide.rst [hunk 20]
```diff
@@ -0,0 +1,209 @@
...
+of reStructuredText, Sphinx can be instructed to parse code comments as well. In fact, in addition
+to what you are reading right now, the `Python documentation <https://www.python.org/doc/>`_
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
to what you are reading right now, the `Python documentation <https://www.python.org/doc/>`__
```

File: docs/developers-guide.rst [hunk 21]
```diff
@@ -0,0 +1,209 @@
...
+has also created using Sphinx.
+
+All of the ``Glue-solar`` documentation is contained in the “docs” folder and code documentation strings.
+
+To build the docs with tox, in the root directory run::
+
+    tox -e build_docs
```
Comment (nabobalis) [inline docs/developers-guide.rst:141]: Is tox setup for glue-solar?

File: docs/developers-guide.rst [hunk 22]
```diff
@@ -0,0 +1,209 @@
...
+
+This command will generate HTML documentation for ``Glue-solar`` in your local "docs/_build/html" directory.
+You can then open the "index.html" file to browse the final docs build.
+
+
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
```

File: docs/developers-guide.rst [hunk 23]
```diff
@@ -0,0 +1,209 @@
...
+Testing Guidelines
+------------------
+
```
Comment (nabobalis) [inline docs/developers-guide.rst:148]:
```suggestion
```

File: docs/developers-guide.rst [hunk 24]
```diff
@@ -0,0 +1,209 @@
...
+Then you can run the tests in any of these doing::
+
+    tox -e <name_of_env>
+
+This will create a test environment in “.tox” and build, install ``Glue-Solar`` and runs the entire test suite.
+This is the method that our continuous integration (CI) uses. Please note that individual unit tests can only be run
+on its own with ``pytest``, with ``tox`` the entire test suite is run per invocation of the command.
+
+
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
```

File: docs/developers-guide.rst [hunk 25]
```diff
@@ -0,0 +1,209 @@
...
+Basic Glue Concepts
+-------------------
+For a thorough treatment of the concepts used in ``Glue``, we recommend going through the official
+documentation specifically for writing custom viewers, which is a three-part series, to be found at
+`Writing a custom viewer for glue <http://docs.glueviz.org/en/latest/customizing_guide/viewer.html>`_,
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
`Writing a custom viewer for glue <http://docs.glueviz.org/en/latest/customizing_guide/viewer.html>`__,
```

File: docs/developers-guide.rst [hunk 26]
```diff
@@ -0,0 +1,209 @@
...
+`Writing a custom viewer for glue with Qt <http://docs.glueviz.org/en/latest/customizing_guide/qt_viewer.html>`_
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
`Writing a custom viewer for glue with Qt <http://docs.glueviz.org/en/latest/customizing_guide/qt_viewer.html>`__
```

File: docs/developers-guide.rst [hunk 27]
```diff
@@ -0,0 +1,209 @@
...
+as well as `Writing a custom viewer for glue with Qt and
+Matplotlib <http://docs.glueviz.org/en/latest/customizing_guide/matplotlib_qt_viewer.html>`_,
```
Comment (nabobalis) [inline docs/developers-guide.rst]:
```suggestion
Matplotlib <http://docs.glueviz.org/en/latest/customizing_guide/matplotlib_qt_viewer.html>`__,
```

File: docs/index.rst [hunk 28]
```diff
@@ -1,13 +1,23 @@
...
+for their solar data.
 
 .. toctree::
-   :maxdepth: 2
+   :maxdepth: 3
 
+   users-guide
+   developers-guide
    loading-aia-and-hmi
    loading-iris-level-2-raster-and-sji-data
```
Comment (nabobalis) [inline docs/index.rst:22]: Later on we will probably want to change the layout of this.

File: docs/users-guide.rst [hunk 29]
```diff
@@ -0,0 +1,72 @@
+.. _users_guide:
+
+============
+User's Guide
+============
+
+Requirements
+------------
+For the requirements of ``Glue``, please refer to their official documentation on
+`installing and running glue <http://docs.glueviz.org/en/stable/installation/installation.html>`_.
```
Comment (nabobalis) [inline docs/users-guide.rst]:
```suggestion
`installing and running glue <http://docs.glueviz.org/en/stable/installation/installation.html>`__.
```

File: docs/users-guide.rst [hunk 30]
```diff
@@ -0,0 +1,72 @@
...
+
+* ``astropy``
+* ``glue-core``
+* ``sunpy``
+* ``ndcube``
+* ``sunraster``
+* ``reproject``
+* ``dask[array]``
+
+
```
Comment (nabobalis) [inline docs/users-guide.rst]:
```suggestion
```

File: docs/users-guide.rst [hunk 31]
```diff
@@ -0,0 +1,72 @@
...
+    cd glue-solar
+
+Finally, do the following while at the root (highest level) of the ``glue-solar`` directory::
+
+    pip install -e .
+
+This is assuming that you already have ``pip``, the Python standard package-management system installed.
+
+
```
Comment (nabobalis) [inline docs/users-guide.rst]:
```suggestion
```

File: docs/users-guide.rst [hunk 32]
```diff
@@ -0,0 +1,72 @@
...
+    glue -v
+
+As per usual, the help ``-h`` option is available for you to get additional information on the usage of glue as well
+as all the options you can use::
+
+    glue -h
+
+On Windows operating systems, installation creates an executable ``glue.exe`` file within the python script directory
+(e.g., C:\Python27\Scripts). Windows users can create a desktop shortcut for this file upon installation,
```
Comment (nabobalis) [inline docs/users-guide.rst]: Python27?

File: docs/users-guide.rst [hunk 33]
```diff
@@ -0,0 +1,72 @@
...
+and run Glue by double clicking on the icon instead.
+
+At this point you should be able to use the functionality offered by both ``Glue`` and ``Glue-solar``.
+
+User Interface Guide
+--------------------
+For a walk through of the user interface of ``Glue``, please refer to the
+`Getting Started <http://docs.glueviz.org/en/stable/getting_started/index.html>`_ section of the official ``Glue``
```
Comment (nabobalis) [inline docs/users-guide.rst]:
```suggestion
`Getting Started <http://docs.glueviz.org/en/stable/getting_started/index.html>`__ section of the official ``Glue``
```

File: docs/developers-guide.rst (hunk 2 above)
Comment (kakirastern) [reply docs/developers-guide.rst]: Sorry I meant `Glue-solar`

File: docs/developers-guide.rst (hunk 4 above)
Comment (kakirastern) [reply docs/developers-guide.rst]: Yup

File: docs/users-guide.rst (hunk 32 above)
Comment (kakirastern) [reply docs/users-guide.rst]: Oh, yeah... Typo. Should be `37`.

File: docs/developers-guide.rst (hunk 1 above)
Comment (kakirastern) [reply docs/developers-guide.rst]: Okay, let me make the changes accordingly

File: docs/index.rst (hunk 28 above)
Comment (kakirastern) [reply docs/index.rst:22]: Yeah, agreed

File: docs/developers-guide.rst (hunk 21 above)
Comment (kakirastern) [reply docs/developers-guide.rst:141]: Yes, I can do `tox -e build_docs` locally at the time being

File: docs/conf.py [hunk 34]
```diff
@@ -25,8 +25,13 @@
     'sphinx.ext.napoleon',
     'sphinx.ext.doctest',
     'sphinx.ext.mathjax',
+    # 'sphinx_automodapi.automodapi',
+    # 'sphinx_automodapi.smart_resolver'
```
Comment (nabobalis) [inline docs/conf.py:29]: Why were these commented out?

**Review by nabobalis — APPROVED** [review]

File: docs/conf.py (hunk 34 above)
Comment (kakirastern) [reply docs/conf.py:29]: Oh, an error was thrown for me locally while trying to run `tox -e build_docs` on my machine, so I comment them out for now.

**Comment by kakirastern** [comment]
Thanks @nabobalis!

**Pull request merged**